### PR TITLE
Some pybridge fixes/cleanups

### PR DESCRIPTION
**Pull request by martinpitt**
Broken out from PR #18757

Fix [this RHEL 8 runtime bridge crash](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18758-20230507-065718-5b53fa4a-centos-8-stream-pybridge/log.html#52) and [this unit test failure during package build](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18757-20230505-203053-94cb8c7a-debian-testing-bots-4733/log.html)

### CODE REVIEW

**Comment by martinpitt** [comment]
[This](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18758-20230507-065718-5b53fa4a-fedora-37-pybridge/log.html#233) is an interesting superuser related pixel failure that is actually a bug. The plumbing of sudo messages is different indeed, and it somehow lost its alert formatting.

File: src/cockpit/superuser.py [hunk 1]
```diff
@@ -82,13 +82,13 @@ def apply_rule(self, options: Dict[str, object]) -> Optional[Peer]:
             raise RoutingError('access-denied')
 
     # ferny.InteractionResponder
-    async def do_askpass(self, messages: str, prompt: str, hint: str) -> Optional[str]:
```
Comment (allisonkarlitskaya) [inline src/cockpit/superuser.py:85]: I generally dislike renaming arguments on virtual function implementations.

You also mention yourself in the dialog that the name "error" might indeed be inaccurate.  We could maybe better call this argument `stderr` but then I'd expect to see a corresponding change in ferny.  For the record, I'd also support moving it to the end of the argument list.

Without changing ferny, I think the thing that makes the most sense is to pass the `messages` argument as the last argument to `self.prompt()` and (possibly) adjust the comment at the site where `prompt` is defined.

**Review by allisonkarlitskaya — CHANGES_REQUESTED** [review]
I'd have let the first one go, but I really don't like the askpass change, sorry...

File: src/cockpit/samples.py [hunk 2]
```diff
@@ -151,11 +151,14 @@ def detect_cpu_sensors(dir_fd: int) -> Iterable[str]:
 
     @staticmethod
     def scan_sensors() -> Iterable[str]:
-        with Handle.open(HWMON_PATH, os.O_RDONLY | os.O_DIRECTORY) as top_fd:
-            for hwmon_name in os.listdir(top_fd):
-                with Handle.open(hwmon_name, os.O_RDONLY | os.O_DIRECTORY, dir_fd=top_fd) as subdir_fd:
-                    for sensor in CPUTemperatureSampler.detect_cpu_sensors(subdir_fd):
-                        yield f'{HWMON_PATH}/{hwmon_name}/{sensor}'
+        try:
```
[the remark this answers is not in the document]
Comment (martinpitt) [reply src/cockpit/samples.py:154]: This is nicer, thanks! Done.

File: src/cockpit/superuser.py (hunk 1 above)
Comment (martinpitt) [reply src/cockpit/superuser.py]: Right, I agree. I reworked it as suggested and added a comment, but not to the declaration (as it's too detached from the place where that "transition" actually happens), but instead to the line below where the prompt signal gets emitted.

**Comment by martinpitt** [comment]
[TestLoopback](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18758-20230508-161455-9bae6c76-centos-8-stream-pybridge/log.html#41-2) fails awfully often on c8s/pybridge. It's a flake, but I'll investigate in the next days.

File: test/common/testlib.py [hunk 3]
```diff
@@ -1366,9 +1366,6 @@ def new_browser(self, machine=None, coverage=False):
             else:
                 if machine.image == reference_image:
                     pixels_label = self.label()
-        # HACK: until @todoPybridge disappears from all pixel tests
-        if os.environ.get("TEST_SCENARIO") == "pybridge":
-            pixels_label = None
```
Comment (allisonkarlitskaya) [inline test/common/testlib.py:1371]: :tada:

File: test/common/testlib.py [hunk 4]
```diff
@@ -1792,15 +1792,16 @@ def check_journal_messages(self, machine=None):
             self.allowed_messages.append("Failed to connect to coredump service: No such file or directory")
             self.allowed_messages.append("Failed to connect to coredump service: Connection refused")
 
+        #
         # HACK: pybridge bugs
-        if os.environ.get("TEST_SCENARIO") == "pybridge":
```
Comment (allisonkarlitskaya) [inline test/common/testlib.py:1799]: I liked the greppability of this, but as you wish...

**Review by allisonkarlitskaya — APPROVED** [review]

File: test/common/testlib.py (hunk 4 above)
Comment (martinpitt) [reply test/common/testlib.py:1799]: Me too -- but you'll see why we have to do this in my next PR, and in #18757

**Pull request merged**